### PR TITLE
Add redux store to project

### PR DESCRIPTION
**Pull request by t-bello7**
- create book reducers, actions and states in bookSlice.js  
- create category reducers, actions and states in categorySlice.js
- create the redux store

### CODE REVIEW

File: src/redux/features/books/bookSlice.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { createSlice } from '@reduxjs/toolkit';
+import { v4 as uuidv4 } from 'uuid';
+
```
Comment (Mihndim2020) [inline src/redux/features/books/bookSlice.js:3]: - [ ] As  it is a [requirement](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/add_reducers_an_actions.md) for this project, " `Define action types for adding and removing a book.` ", please kindly define your action types, and the value of each action type should follow the [ducks pattern](https://github.com/erikras/ducks-modular-redux). Some like so(`just an example`): 
```
const ADD_BOOK = micro-bookstore/books/ADD_BOOK
const REMOVE_BOOK = micro-bookstore/books/REMOVE_BOOK
```

File: src/redux/features/categories/categorySlice.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { createSlice } from '@reduxjs/toolkit';
+
```
Comment (Mihndim2020) [inline src/redux/features/categories/categorySlice.js:2]: - [ ] As  it is a [requirement](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/add_reducers_an_actions.md) for this project, " `Define an action type for checking the status.` ", please kindly define your action types, and the value of each action type should follow the [ducks pattern](https://github.com/erikras/ducks-modular-redux). Some like so(`just an example`): 
```
const const CHECK_STATUS = micro-bookstore/books/CHECK_STATUS

```

**Review by Mihndim2020 — CHANGES_REQUESTED** [review]
### Hi @t-bello7, 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Great job with RTK
:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

There are some issues that you still need to work on to get this project approved, but you are almost there!
ALMOST-MINION


# Status: Requires Changes ♻️
_Check the comments under the review._


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/features/books/bookSlice.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        producer: action.payload.producer,
+      });
+    },
+    remove(state, action) {
+      state.value.filter((book) => action.id !== book.id);
+    },
+  },
+});
+
+export const { add, remove } = BookSlice.actions;
```
Comment (t-bello7) [inline src/redux/features/books/bookSlice.js:26]: Thank you for the review @Mihndim2020 however the actions are automatically created on this line when using the createSlice method.

File: src/redux/features/books/bookSlice.js (hunk 1 above)
Comment (t-bello7) [reply src/redux/features/books/bookSlice.js:3]: @Mihndim2020 The add and remove books actions are automatically created on line 26 following the DUCK principle. You can check out the official [redux-toolkit]( https://redux-toolkit.js.org/tutorials/quick-start) documentation to learn more about it.

File: src/redux/features/categories/categorySlice.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  name: 'category',
+  initialState,
+  reducers: {
+    check() {
+      return 'Under construction';
+    },
+  },
+});
+
+export const { check } = categorySlice.actions;
```
Comment (t-bello7) [inline src/redux/features/categories/categorySlice.js:17]: @Mihndim2020  The check action is created on line 17.

File: src/redux/features/books/bookSlice.js (hunk 1 above)
Comment (t-bello7) [reply src/redux/features/books/bookSlice.js:3]: Thank you @Mihndim2020  for the code review. The actions are automatically created when using the createSlice method.

**Review by Davidosky007 — APPROVED** [review]
Hi @t-bello7 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* No Linter errors 👍 
* Well written readme file 👍 
* Descriptive PR description 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**